tests/test_code.py

```diff
@@ -0,0 +1,9 @@
+import os
```

Comment:
Would be good to include this test in the workflow somehow. E.g. could add [here](https://github.com/pqzx/html2docx/blob/b9d30ceca5da5cc6e398b03832f1dd3448e8b3aa/.github/workflows/python-app.yml#L35) (and include the artifact in test outputs).

Reply:
That's a good idea. I'll try and add that over the next few days.

Reply:
I ended up making a few changes:

- Add unittests to test code block, pre block and inline use of code block.
- Made a change to `remove_whitespace` when used for code blocks. This prevents a space character being inserted at the beginning of code blocks.
- Added doctests to demonstrate how `remove_whitespace` works. This is more for clarity than anything, but minor changes to the function could cause big changes to how things are rendered. Adding doctests and running them in the workflow will ensure any changes don't break anything.

I still need to update the workflow ([can't do it over an https clone?!](https://stackoverflow.com/a/58885125/6252525)), but I'm planning to add this to run the `test_code` test module and the doctests in `h2d`.

```
        python -m tests.test_code
        python -m doctest -v htmldocx/h2d.py
```

The outputs end up looking like this:

> ![image](https://user-images.githubusercontent.com/30441316/127009940-0ce9bc0c-0a7c-466c-b21e-cdc1110253d8.png)

> ![image](https://user-images.githubusercontent.com/30441316/127009993-60590d1a-d9eb-42e6-93cc-ae34e8f48426.png)

> ![image](https://user-images.githubusercontent.com/30441316/127010049-ccfabf02-8e5f-4f32-b596-11e591ba1b84.png)

Note the difference from the previous version that ended up with the extra white space:

> ![image](https://user-images.githubusercontent.com/30441316/127010347-884e520f-9b18-4055-aac7-1b6e1f973405.png)

Reply:
Great work on this. Thanks for the clear explanation 😄